src/pages/namespace/Instances/Detail/store/instanceContext.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  FC,
+  PropsWithChildren,
+  createContext,
+  useContext,
+  useReducer,
+} from "react";
+import { Actions as InstanceActions, StateType } from "./type";
+
+const InstanceStateContext = createContext<StateType | null>(null);
+
+const InstanceDispatchContext =
+  createContext<React.Dispatch<InstanceActions> | null>(null);
+
+function stateReducer(state: StateType, action: InstanceActions) {
+  switch (action.type) {
+    case "UPDATE_FILTER_STATE_NAME": {
+      const newState: StateType = {
+        ...state,
+        filters: {
+          ...state.filters,
+          workflowName: state.filters?.workflowName ?? undefined,
+          stateName: action.payload.stateName,
+        },
+      };
+      return newState;
+    }
+    case "UPDATE_FILTER_WORKFLOW": {
+      const newState: StateType = {
+        ...state,
+        filters: {
+          ...state.filters,
+          workflowName: action.payload.workflowName,
+          stateName: state.filters?.stateName ?? undefined,
+        },
+      };
+      return newState;
+    }
+
+    default: {
+      return state;
+    }
+  }
+}
+
+const Provider: FC<PropsWithChildren & { instanceId: string }> = ({
+  children,
+  instanceId,
+}) => {
+  const [state, dispatch] = useReducer(stateReducer, { instanceId });
+
+  return (
+    <InstanceStateContext.Provider value={state}>
+      <InstanceDispatchContext.Provider value={dispatch}>
+        {children}
+      </InstanceDispatchContext.Provider>
+    </InstanceStateContext.Provider>
+  );
+};
+
+const useInstanceId = () => {
+  const context = useContext(InstanceStateContext);
+  if (!context) {
+    throw new Error("useFilter must be used within a InstanceContext");
+  }
+
+  return context.instanceId;
+};
+
+const useFilters = () => {
+  const context = useContext(InstanceStateContext);
+  if (!context) {
+    throw new Error("useFilter must be used within a InstanceContext");
```

Comment:
```suggestion
    throw new Error("useFilters must be used within a InstanceContext");
```
If I understand correctly this should match the method name?

Reply:
That's right, I updated it for both methods (ad5fefdf02bb153abb8e7d140f4a9b99b764867a)